cvmfs/util/algorithm.cc

```diff
@@ -69,6 +69,131 @@ double StopWatch::GetTime() const {
+  {
+    max_left_boundary_count = max(max_left_boundary_count,
+                                count_digits(boundary_values[i] / 2));
+    max_right_boundary_count = max(max_right_boundary_count,
+                                count_digits(boundary_values[i] - 1));
+    max_value_count = max(max_value_count, count_digits(bins[i]));
+    max_bins = max(max_bins, (uint)atomic_read32(&bins[i]));
+    total_sum_of_bins += (uint)atomic_read32(&bins[i]);
+  }
+
+  max_bins = max(max_bins, (uint)atomic_read32(&bins[0]));
+  total_sum_of_bins += (uint)atomic_read32(&bins[0]);
+
+
+  max_stars = max_bins * total_stars / total_sum_of_bins;
+
+  string format = " %" + to_string(max_left_boundary_count < 2 ?
+                                  2 : max_left_boundary_count) +
+                  "d -> %" + to_string(max_right_boundary_count) +
+                  "d :     %" + to_string(max_value_count) + "d | %" +
+                  to_string(max_stars) + "s |\n";
+
+  string title_format = " %" +
+                  to_string((max_left_boundary_count < 2 ?
+                              2 : max_left_boundary_count) +
+                              max_right_boundary_count +
+                              4) +
+                  "s | %" + to_string(max_value_count + 4) +
+                  "s | %" + to_string(max_stars) + "s |\n";
+
+  string overflow_format = "%" +
+                  to_string(max_left_boundary_count +
+                              max_right_boundary_count +
+                              5) +
+                  "s : %" + to_string(max_value_count + 4) +
+                  "d | %" + to_string(max_stars) + "s |\n";
+
+  string result_string = "";
+
+  char buffer[BUFFSIZE];
+  memset(buffer, 0, sizeof(buffer));
+
+  snprintf(buffer,
+      BUFFSIZE,
+      title_format.c_str(),
+      "usec",
+      "count",
+      "distribution");
+  result_string += buffer;
+  memset(buffer, 0, sizeof(buffer));
+
+  for (i = 1; i <= number_of_bins; i++)
+  {
+    uint n_of_stars = (uint)atomic_read32(&bins[i]) * total_stars /
+                                                  total_sum_of_bins;
+    snprintf(buffer,
+            BUFFSIZE,
+            format.c_str(),
+            boundary_values[i - 1],
+            boundary_values[i] - 1,
+            (uint)atomic_read32(&bins[i]),
+            generate_stars(n_of_stars).c_str());
+    result_string += buffer;
+    memset(buffer, 0, sizeof(buffer));
+  }
+
+  uint n_of_stars = (uint)atomic_read32(&bins[0]) * total_stars /
+                                                  total_sum_of_bins;
+  snprintf(buffer,
+          BUFFSIZE,
+          overflow_format.c_str(),
+          "overflow",
+          (uint)atomic_read32(&bins[0]),
+          generate_stars(n_of_stars).c_str());
+  result_string += buffer;
+  memset(buffer, 0, sizeof(buffer));
+
+  return result_string;
+}
+
+void Log2Hist::PrintLog2Hist()
+{
+  printf("%s", this->Print().c_str());
+  fflush(stdout);
```

Comment:
Not needed if the printed string ends with a newline.

Reply:
Exactly! My bad.